### DISCUSSION TITLE
Alternative way to add components

### DISCUSSION THREAD

**Discussion by stephane-vanraes** [svelte-society/sveltesociety.dev#318] [Ideas]
The current way of adding community resources to `components.json` is cumbersome and prone to merge conflicts.  We can use this thread to brainstorm some alternative approaches (and maybe come to the conclusion to keep everything the way it is).

to keep things organized, make a "thread" for each idea.

**Comment by stephane-vanraes** [comment]
One idea is to grab the data from a database:

pros
+ components can be added through a form
+ no mess with users inventing new tags (only predefined ones in a select possible)

cons
- not possible to statically render the page anymore
- hosting ?
- still needs a system to curate and approve

**Comment by MacFJA** [comment]
Another idea is to use something like a Google Form:
- The form will fill a Google Sheet.
- Reviewers will fix and validate data
- Validated data will be copied/transfer to the JSON files (this step can be automated)

---

### Pros

- Easy to add new content of user
- Don't change how the website get/read data

### Cons

- Manual labor to validate the content
- Need to setup a form and a sheet
- Google
  - Need a Google account to setup the form
  - Need a Google account to fill the form
  - Google is not Open Source

**Comment by MacFJA** [comment]
Suggestion made in the PR #393 

----

Make every component, template, tool into its own file

----

### Pros

- No more merge conflict

### Cons

- Lots of files
- Harder to check data consistency (same tags, category across all items)
- Naming convention for filename to create and enforce
  - Are spaces allowed ?
  - Are `+` allowed ?
  - Are `@` allowed ?
  - Are `/` allowed ?
  - How to manage name collision if some chars are excluded

**Comment by webJose** [comment]
Suggestion:

Add a "temporary" folder where people add their contributions as a separate JSON file.  Then create a GitHub action that merges the individual JSON file from the "temporary" folder into the current JSON file, after running all checks.  Have the action submit the change.  This should avoid merge conflicts as long as people don't get into name collisions.  Perhaps a file name generator can be volunteered?

## Pros

+ No merge conflicts
+ Once the merge of the JSON's is done, the current flow continues as is

## Cons

+ No idea about the internals of how things work.  Kindly someone fill this up.
